src/Conversion/ONNXToTOSA/Math/Elementwise.cpp

```diff
@@ -57,25 +75,34 @@ class ONNXBinaryElementwiseOpLoweringToTOSA
   LogicalResult matchAndRewrite(ONNXOpT op, OpAdaptor adaptor,
       ConversionPatternRewriter &rewriter) const override {
 
-    Value lhs = adaptor.A();
-    auto lhsType = lhs.getType().dyn_cast<TensorType>();
+    if (failed(checkBasicTosaRequirementsForBinaryOps<OpAdaptor>(
+            rewriter, op, adaptor, op.getResult().getType())))
+      return failure();
 
+    Value lhs = adaptor.A();
     Value rhs = adaptor.B();
-    auto rhsType = rhs.getType().dyn_cast<TensorType>();
 
-    auto resultType = op.getResult().getType().template dyn_cast<TensorType>();
-    if (!lhsType || !rhsType || !resultType) {
-      return rewriter.notifyMatchFailure(op, "Tosa only supports TensorTypes");
-    }
+    rewriter.replaceOpWithNewOp<TosaOpT>(op, op.getType(), lhs, rhs);
 
-    Type resultElementType = resultType.getElementType();
+    return success();
+  }
+};
 
-    if (!resultElementType.isIntOrFloat()) {
-      return rewriter.notifyMatchFailure(
-          op, "only int and float are supported");
-    }
+class ONNXMulOpLoweringToTosa : public OpConversionPattern<ONNXMulOp> {
+public:
+  using OpConversionPattern::OpConversionPattern;
+  LogicalResult matchAndRewrite(ONNXMulOp op, OpAdaptor adaptor,
+      ConversionPatternRewriter &rewriter) const override {
+    if (failed(checkBasicTosaRequirementsForBinaryOps<OpAdaptor>(
+            rewriter, op, adaptor, op.getResult().getType())))
+      return failure();
 
-    rewriter.replaceOpWithNewOp<TosaOpT>(op, op.getType(), lhs, rhs);
+    Value lhs = adaptor.A();
+    Value rhs = adaptor.B();
+
+    // Only shift 0 is supported for now.
```

Comment:
The comment seems inappropriate - `onnx.mul` doesn't include a shift specifier for integer multiplication, so shift must be zero for this conversion rather than being a question of supporting it or not. (I guess the shift could come if the output was going to an appropriate `onnx.bitshift`, but we'd more likely depend on a tosa optimization pass to perform this simplification.)